coderd/runtimeconfig/config.go

```diff
@@ -0,0 +1,97 @@
+package runtimeconfig
+
+import (
+	"context"
+	"errors"
+	"reflect"
+
+	"github.com/spf13/pflag"
+	"golang.org/x/xerrors"
+)
+
+// TODO: comment
+type Value pflag.Value
+
+type Entry[T Value] struct {
+	val T
+	key string
+}
+
+func New[T Value](key, val string) (out Entry[T], err error) {
+	out.Init(key)
+
+	if err = out.Set(val); err != nil {
+		return out, err
+	}
+
+	return out, nil
+}
+
+func MustNew[T Value](key, val string) Entry[T] {
+	out, err := New[T](key, val)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+func (o *Entry[T]) Init(key string) {
+	o.val = create[T]()
```

Comment:
Is this to prevent a nil derefence? Why do we need to `init` the value?

Reply:
I've adjusted the implementation to handle zero values more elegantly.